workbench/src/menu.c

```diff
@@ -0,0 +1,310 @@
+		menu_item_close_activate();
+		sidebar_update(SIDEBAR_CONTEXT_WB_CREATED, NULL);
+	}
+	else
+	{
+		dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not create new workbench file: %s"), error->message);
+		workbench_free(wb_globals.opened_wb);
+		wb_globals.opened_wb = NULL;
+	}
+	g_free(filename);
+}
+
+
+/* The function handles the menu item "Open workbench" */
+static void item_open_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	gchar *filename;
+	GError *error = NULL;
+
+	filename = dialogs_open_workbench();
+	if (filename == NULL)
+	{
+		return;
+	}
+	wb_globals.opened_wb = workbench_new();
+	if (workbench_load(wb_globals.opened_wb, filename, &error))
+	{
+		menu_item_new_deactivate();
+		menu_item_open_deactivate();
+		menu_item_save_activate();
+		menu_item_settings_activate();
+		menu_item_close_activate();
+		sidebar_update(SIDEBAR_CONTEXT_WB_OPENED, NULL);
+	}
+	else
+	{
+		dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not open workbench file: %s"), error->message);
+		workbench_free(wb_globals.opened_wb);
+		wb_globals.opened_wb = NULL;
+	}
+	g_free(filename);
+}
+
+
+/* The function handles the menu item "Save workbench" */
+static void item_save_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	GError *error = NULL;
+
+	if (wb_globals.opened_wb != NULL)
+	{
+		if (!workbench_save(wb_globals.opened_wb, &error))
+		{
+			dialogs_show_msgbox(GTK_MESSAGE_INFO, _("Could not save workbench file: %s"), error->message);
+		}
+		sidebar_update(SIDEBAR_CONTEXT_WB_SAVED, NULL);
+	}
+}
+
+
+/* The function handles the menu item "Settings" */
+static void item_workbench_settings_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	if (wb_globals.opened_wb != NULL)
+	{
+		if (dialogs_workbench_settings(wb_globals.opened_wb))
+		{
+			sidebar_update(SIDEBAR_CONTEXT_WB_SETTINGS_CHANGED, NULL);
+		}
+	}
+}
+
+
+/* The function handles the menu item "Close workbench" */
+static void item_close_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	workbench_free(wb_globals.opened_wb);
+	wb_globals.opened_wb = NULL;
+
+	menu_item_new_activate();
+	menu_item_open_activate();
+	menu_item_save_deactivate();
+	menu_item_settings_deactivate();
+	menu_item_close_deactivate();
+
+	sidebar_update(SIDEBAR_CONTEXT_WB_CLOSED, NULL);
+}
+
+
+/** Setup the workbench menu.
+ *
+ **/
+gboolean menu_init(void)
+{
+	/* Create menu and root item/label */
+	menu_data.menu = gtk_menu_new();
+	menu_data.root_item = gtk_menu_item_new_with_label(_("Workbench"));
+	gtk_widget_show(menu_data.root_item);
+
+	/* Create new menu item "New Workbench" */
+	menu_data.item_new = gtk_menu_item_new_with_mnemonic(_("New"));
+	gtk_widget_show(menu_data.item_new);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_new);
+	g_signal_connect(menu_data.item_new, "activate",
+					 G_CALLBACK(item_new_workbench_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_new, NULL);
```

Comment:
Hmm...removed the calls. I do not remember exactly. Maybe copied from some place. I also thought I can use it to get the items destroyed on ending geany/the plugin.